hazelcast/src/test/java/com/hazelcast/test/backup/CacheBackupAccessor.java

```diff
@@ -0,0 +1,203 @@
+
+import com.hazelcast.cache.HazelcastCacheManager;
+import com.hazelcast.cache.ICache;
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.cache.impl.HazelcastServerCacheManager;
+import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
+import com.hazelcast.cache.impl.ICacheRecordStore;
+import com.hazelcast.cache.impl.record.CacheRecord;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.partition.InternalPartition;
+import com.hazelcast.internal.partition.InternalPartitionService;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.partition.IPartition;
+import com.hazelcast.spi.serialization.SerializationService;
+
+import javax.cache.expiry.ExpiryPolicy;
+import javax.cache.spi.CachingProvider;
+
+import static com.hazelcast.test.HazelcastTestSupport.getNode;
+import static com.hazelcast.test.HazelcastTestSupport.getNodeEngineImpl;
+import static com.hazelcast.test.TestTaskExecutorUtil.runOnPartitionThread;
+
+/**
+ * Implementation of {@link BackupAccessor} for {@link ICache}.
+ *
+ * @param <K> type of keys
+ * @param <V> type of values
+ */
+class CacheBackupAccessor<K, V> extends AbstractBackupAccessor<K, V> implements BackupAccessor<K, V> {
+
+    private final String cacheName;
+
+    CacheBackupAccessor(HazelcastInstance[] cluster, String cacheName, int replicaIndex) {
+        super(cluster, replicaIndex);
+        this.cacheName = cacheName;
+    }
+
+    @Override
+    public int size() {
+        InternalPartitionService partitionService = getNode(cluster[0]).getPartitionService();
+        IPartition[] partitions = partitionService.getPartitions();
+        int count = 0;
+        for (IPartition partition : partitions) {
+            Address replicaAddress = partition.getReplicaAddress(replicaIndex);
+            if (replicaAddress == null) {
+                continue;
+            }
+
+            HazelcastInstance hz = getInstanceWithAddress(replicaAddress);
+            CachingProvider provider = HazelcastServerCachingProvider.createCachingProvider(hz);
+            HazelcastCacheManager cacheManager = (HazelcastServerCacheManager) provider.getCacheManager();
+
+            NodeEngineImpl nodeEngine = getNodeEngineImpl(hz);
+            CacheService cacheService = nodeEngine.getService(CacheService.SERVICE_NAME);
+            String cacheNameWithPrefix = cacheManager.getCacheNameWithPrefix(cacheName);
+            int partitionId = partition.getPartitionId();
+
+            count += runOnPartitionThread(hz, new SizeCallable(cacheService, cacheNameWithPrefix, partitionId), partitionId);
+        }
+        return count;
+    }
+
+    @Override
+    public V get(K key) {
+        IPartition partition = getPartitionForKey(key);
+        HazelcastInstance hz = getHazelcastInstance(partition);
+
+        Node node = getNode(hz);
```

Comment:
I did some general cleanup to call `getNode()` and `getNodeEngineImpl()` just once, otherwise we'll create multiple proxy classes when used by the `HazelcastStarter` framework. The logic of the `BackupAccessor` implementations is still the same, it's just more `HazelcastStarter` friendly now.